LocalPackages/DataBrokerProtection/Tests/DataBrokerProtectionTests/DataBrokerProfileQueryOperationManagerTests.swift

```diff
@@ -389,6 +396,8 @@ final class DataBrokerProfileQueryOperationManagerTests: XCTestCase {
             XCTAssertTrue(mockDatabase.optOutEvents.contains(where: { $0.type == .optOutConfirmed }))
             XCTAssertTrue(mockDatabase.wasUpdateRemoveDateCalled)
             XCTAssertNotNil(mockDatabase.extractedProfileRemovedDate)
+            XCTAssertTrue(mockDatabase.wasUpdatedPreferredRunDateForOptOutCalled)
```

Comment:
This assertion disappeared some time in the last year. Added it back to check for the correct preferred run date.

Reply:
I'm worried this was removed deliberately. I do remember at one point the deliberate choice to keep doing certain operations for old data even when a user had changed their details.

Reply:
@jotaemepereira/@Bunn Do you have any context on this?

Reply:
@quanganhdo, I do not have the full context of the change; I need to dig deeper. What @THISISDINOSAUR is mentioning is true.

When a user changes their profile and there are current opt-outs in progress for the old profile, we will:
- Not scan for new matches in the old profile
- In-progress matches for the old profile should be driven to completion (opt-out and then confirmation)

I do not know if this helps; ping me on MM if you need more assistance.